compiler/src/iree/compiler/Codegen/Common/ReshapePatterns.cpp

```diff
@@ -587,41 +587,79 @@ struct FoldCollapseShapeIntoInterfaceTensorStore
                                 PatternRewriter &rewriter) const override {
     // Bail out if the strides aren't unit.
     if (!llvm::all_of(storeOp.getMixedStrides(), isOneInteger)) {
-      return failure();
+      return rewriter.notifyMatchFailure(storeOp, "expected unit strides");
     }
 
-    auto collapseShape =
-        storeOp.getValue().getDefiningOp<tensor::CollapseShapeOp>();
-    // TODO: Support dynamic shapes.
-    if (!collapseShape || !collapseShape.getSrcType().hasStaticShape()) {
-      return failure();
+    auto collapseShape = dyn_cast_if_present<tensor::CollapseShapeOp>(
+        storeOp.getValue().getDefiningOp());
+    if (!collapseShape) {
+      return rewriter.notifyMatchFailure(
+          storeOp, "expected `tensor.collapse_shape` source");
+    }
+    if (collapseShape.getType().getRank() !=
+        storeOp.getTargetType().getRank()) {
+      return rewriter.notifyMatchFailure(
+          storeOp, "expected `tensor.collapse_shape` rank to be the same as "
+                   "the `iree_tensor_ext.dispatch.tensor.store` rank.");
     }
+    RankedTensorType reshapeSrcType = collapseShape.getSrcType();
+    ArrayRef<int64_t> reshapeSrcShape = reshapeSrcType.getShape();
 
     auto subspanOp =
         storeOp.getTarget()
             .template getDefiningOp<IREE::HAL::InterfaceBindingSubspanOp>();
-    // TODO: Support dynamic dims.
-    if (!subspanOp || !subspanOp.getDynamicDims().empty()) {
-      return failure();
+    if (!subspanOp) {
+      return rewriter.notifyMatchFailure(
+          storeOp, "expected `hal.interface.binding.subspan` target");
     }
 
     auto subspanType =
         llvm::cast<IREE::TensorExt::DispatchTensorType>(subspanOp.getType());
-
-    ArrayRef<int64_t> reshapeSrcShape = collapseShape.getSrcType().getShape();
+    SmallVector<Value> dynamicDims = subspanOp.getDynamicDims();
 
     // Verify the subspan shape against the shape of the slice being inserted.
-    for (auto [size, group] : llvm::zip_equal(
-             subspanType.getShape(), collapseShape.getReassociationIndices())) {
-      if (group.size() == 1) {
+    size_t dynIndex = 0;
+    for (auto &&[i, values] : llvm::enumerate(llvm::zip_equal(
+             subspanType.getShape(), collapseShape.getReassociationIndices(),
+             storeOp.getMixedSizes()))) {
+      auto [subspanSize, group, size] = values;
+      // Check that each reassociation group contains at most a single dynamic
+      // dimension.
+      unsigned numDynamicDims =
+          llvm::count_if(group, [reshapeSrcShape](int64_t dim) {
+            return ShapedType::isDynamic(reshapeSrcShape[dim]);
+          });
+      if (numDynamicDims > 1) {
+        return rewriter.notifyMatchFailure(
+            collapseShape,
+            "Got multiple dynamic dimensions in group: " + std::to_string(i));
```

Comment:
```suggestion
            "got multiple dynamic dimensions in group: " + std::to_string(i));
```

> Also, to match error message styles commonly produced by other tools, start the first sentence with a lower-case letter

https://llvm.org/docs/CodingStandards.html#error-and-warning-messages

Same comment for the below notifyMatchFailure.

Reply:
Updated them